crates/ruff/src/rules/pydocstyle/rules/blank_before_after_class.rs

```diff
@@ -228,8 +228,33 @@ pub(crate) fn blank_before_after_class(checker: &mut Checker, docstring: &Docstr
         }
 
         let mut lines = UniversalNewlineIterator::with_offset(after, after_range.start());
+        let first_line = lines.next();
+        let first_line_start = first_line.as_ref().map(|l| l.start()).unwrap_or_default();
+
+        // Edge case: The next statement after the docstring is on the same line separated by a
+        // semicolon.
+        if let Some(first_line) = &first_line {
+            if !first_line.trim().is_empty() {
+                let next_statement = after
+                    .trim_start()
```

Comment:
@MichaReiser since this is not indentation but whitespace between statement and a following `;` i don't need to use the python whitespace functions, right?

Reply:
It’s best to use the Python whitespace functions unless you’re parsing text within a string or within a comment (is my understanding).

Reply:
From the python spec

> Except at the beginning of a logical line or in string literals, the whitespace characters space, tab and formfeed can be used interchangeably to separate tokens. Whitespace is needed between two tokens only if their concatenation could otherwise be interpreted as a different token (e.g., ab is one token, but a b is two tokens).

[source](https://docs.python.org/3/reference/lexical_analysis.html#whitespace-between-tokens)

To me this means that any other whitespace should not be considered whitespace and are a syntax error (because whitespace isn't a valid identifier start). 

I use the python whitespace methods consistently to:

* Avoid having to reason about whether it is guaranteed that the input program is correct and thus, it is safe to accept all whitesapce
* `is_python_whitespace` (or the trim variants) could in theory be faster because they can operate on bytes (no unicode)